test/Motor.Extensions.Hosting.RabbitMQ_UnitTest/RabbitMQMessageHostBuilderExtensionsTests.cs

```diff
@@ -0,0 +1,136 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Moq;
+using Motor.Extensions.Conversion.Abstractions;
+using Motor.Extensions.Hosting.Abstractions;
+using Motor.Extensions.Hosting.CloudEvents;
+using Motor.Extensions.Hosting.Consumer;
+using Motor.Extensions.Hosting.Publisher;
+using Motor.Extensions.Hosting.RabbitMQ;
+using Motor.Extensions.Utilities;
+using Motor.Extensions.Utilities.Abstractions;
+using Xunit;
+
+namespace Motor.Extensions.Hosting.RabbitMQ_UnitTest
+{
+    public class RabbitMQMessageHostBuilderExtensionsTests
+    {
+        private static IConfiguration GetConfigWithVHost(string vHost)
+        {
+            var configDict = new Dictionary<string, string>
+            {
+                { "Host", "localhost" },
+                { "User", "guest" },
+                { "Password", "guest" },
+                { "Queue:Name", "Test" },
+                { "VirtualHost", vHost },
+                { "PublishingTarget:RoutingKey", "routing.key" },
+                { "PublishingTarget:Exchange", "amq.topic" }
+            };
+            return new ConfigurationBuilder().AddInMemoryCollection(configDict).Build();
+        }
+
+        private static IMotorHostBuilder GetHostBuilder()
+        {
+            return MotorHost.CreateDefaultBuilder()
+                .ConfigureSingleOutputService<string, string>()
+                .ConfigureServices((_, services) =>
+                {
+                    services.AddTransient(_ =>
+                    {
+                        var mock = new Mock<IApplicationNameService>();
```

Comment:
mini nit: why not just implement the interface and add an instance:

```cs
record AppNameMock(string Version, string LibVersion, string FullName, Uri Source) : IApplicationNameService
        {
            public string GetVersion() => Version;
            public string GetLibVersion() => LibVersion
            public string GetFullName() => FullName;
            public Uri GetSource() => Source
        }
```

IMHO mocks are most of the time harder to read than normal code but it's up to you

btw: weird interface for C# :sweat_smile: we could think of a refactoring there...

Reply:
I would also prefer this if we were using the mock in multiple places. But since it's only one declaration, I prefer to leave it as is.

Reply:
Actually there are a lot of mocked usages of the `IApplicationNameService` :smile: but that's probably something for a cleanup PR later on